service/slice_config_webhook_validation.go

```diff
@@ -107,6 +112,43 @@ func ValidateSliceConfigDelete(ctx context.Context, sliceConfig *controllerv1alp
 	return nil
 }
 
+func validateCertsRotationInterval(ctx context.Context, sliceConfig *controllerv1alpha1.SliceConfig, old runtime.Object) *field.Error {
+	oldSliceConfig := old.(*controllerv1alpha1.SliceConfig)
+	// nochange detected
+	if sliceConfig.Spec.RenewBefore == oldSliceConfig.Spec.RenewBefore || sliceConfig.Spec.RenewBefore.IsZero() {
+		return nil
+	}
+	// change detected
+	vpnkeyrotaion := controllerv1alpha1.VpnKeyRotation{}
+	exists, _ := util.GetResourceIfExist(ctx, types.NamespacedName{
+		Namespace: sliceConfig.Namespace,
+		Name:      sliceConfig.Name,
+	}, &vpnkeyrotaion)
+	if exists {
+		for cluster := range vpnkeyrotaion.Status.CurrentRotationState {
```

Comment:
I think vpnkeyrotaion.Status.CurrentRotationState shall give the map of gateways right not clusters ?

Reply:
right!